src/chain/ethereum.rs

```diff
@@ -0,0 +1,114 @@
+//! Ethereum
+
+use alloy_chains::NamedChain;
+use alloy_consensus::TxType;
+use alloy_primitives::U256;
+use alloy_rpc_types::{Header, Transaction};
+use revm::primitives::SpecId;
+
+use super::PevmChain;
+
+/// Implementation of [PevmChain] for Ethereum
+#[derive(Debug, Clone, PartialEq)]
+pub struct PevmEthereum {
+    id: u64,
+}
+
+impl PevmEthereum {
+    /// Ethereum Mainnet
+    pub fn mainnet() -> Self {
+        Self {
+            id: NamedChain::Mainnet.into(),
+        }
+    }
+
+    // TODO: support Ethereum Sepolia and other testnets
+}
+
+/// Error type for [PevmEthereum::get_block_spec].
+#[derive(Debug, Clone, PartialEq)]
+pub enum GetBlockSpecError {
+    /// When [header.number] is none.
+    MissingBlockNumber,
+    /// When [header.total_difficulty] is none.
+    MissingTotalDifficulty,
+}
+
+/// Error type for [PevmEthereum::get_gas_price].
+#[derive(Debug, Clone, PartialEq)]
+pub enum GetGasPriceError {
+    /// [tx.type] is invalid.
+    InvalidType(u8),
+    /// [tx.gas_price] is none.
+    MissingGasPrice,
+    /// [tx.max_fee_per_gas] is none.
+    MissingMaxFeePerGas,
+}
+
+impl PevmChain for PevmEthereum {
+    type GetBlockSpecError = GetBlockSpecError;
+    type GetGasPriceError = GetGasPriceError;
```

Comment:
Looking back, `BlockGasPriceError = EthereumGasPriceError` is clearer.

Reply:
Gas price is a property of a tx, not a block.

How about `TxGasPriceError`?

Reply:
Ah yes, just `GasPriceError` is good. Was a copy-paste mistake from `BlockSpecError`.